feat(hub): the long and winding road to 1 + 1 = 2

letypequividelespoubelles opened the pull request:
This class contains peripheral information of the transaction required for the tracing.

OlivierBBB commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/Hub.java:
```diff
@@ -652,7 +652,7 @@ public int contextNumberNew(WorldView world) {
         || opCode.getData().instructionFamily().equals(InstructionFamily.HALT)
         || opCode.getData().instructionFamily().equals(InstructionFamily.INVALID)) {
-      return this.callStack().caller().contextNumber();
+      return this.callStack.depth() == 0 ? 0 : this.callStack().caller().contextNumber(); //TODO: when callStackDepth = 0
```
In the arithmetization we have a shared column called CALLER_CN (if I'm not misremembering the name) which is context-constant etc. In any case, this number will always contain the proper parent execution context. In case you are the root context you really have 2 parent context numbers:
- CALLER_CONTEXT_NUMBER (shared column, the one I mentioned above)
- context/CALL_DATA_CONTEXT_NUMBER (column in the context perspective, which points to the special execution context number whose RAM contains the transaction call data.)

lorenzogentile404 committed:
fix: simplified SELFDESTRUCT handling

lorenzogentile404 committed:
feat(selfdestruct): managed part before transaction is completed

letypequividelespoubelles committed:
feat(hub): done with exceptional, aborting, failure, emptyInit CREATE(s)

letypequividelespoubelles committed:
fix(hub): regenerate Trace.java

letypequividelespoubelles committed:
feat(hub): CREATE ready for testing and debuging

OlivierBBB committed:
ras

letypequividelespoubelles committed:
test: add CREATE test

lorenzogentile404 committed:
feat(selfdestruct): managed will revert case

OlivierBBB committed:
feat(return): wip

letypequividelespoubelles committed:
feat(return): implement mmu call invalid code prefix

lorenzogentile404 committed:
feat(selfdestruct): review and complete will revert case

lorenzogentile404 committed:
feat(selfdestruct): added comments

lorenzogentile404 committed:
ras

OlivierBBB committed:
ras

letypequividelespoubelles committed:
clean stuff

lorenzogentile404 committed:
feat(selfdestruct): started implementing self destruct map

lorenzogentile404 committed:
feat(selfdestruct): rename components of the map

lorenzogentile404 committed:
feat(selfdestruct): implement resolvePostTransaction

OlivierBBB committed:
feat(return): progress but some TODO's

letypequividelespoubelles committed:
delete the failure signals, delete romLex signals, triggers romLex for CREATE, seal sections for CN/CNnew

letypequividelespoubelles committed:
Merge branch 'arith-dev' into feat/issue-746/linea-transaction-class

letypequividelespoubelles committed:
merge arith-dev

letypequividelespoubelles committed:
rm void file

lorenzogentile404 committed:
feat(selfdestruct): cleaned code

letypequividelespoubelles committed:
Merge branch 'arith-dev' into feat/issue-746/linea-transaction-class

letypequividelespoubelles committed:
fix resolve defers after triggering module at endTx

letypequividelespoubelles committed:
fix: need to do two rounds of defersPostTx

letypequividelespoubelles committed:
Merge branch 'arith-dev' into feat/issue-746/linea-transaction-class

letypequividelespoubelles committed:
Merge branch 'arith-dev' into feat/issue-746/linea-transaction-class

letypequividelespoubelles committed:
extcodecopy triggers RomLex

letypequividelespoubelles committed:
Merge branch 'feat/issue-746/linea-transaction-class' of github.com:Consensys/linea-arithmetization into feat/issue-746/linea-transaction-class

lorenzogentile404 committed:
feat(selfdestruct): fixed resolvePostTransaction

letypequividelespoubelles committed:
fix(return): triggers ROMLEX

OlivierBBB committed:
fix(stuff): various

letypequividelespoubelles committed:
spotless + clean

lorenzogentile404 committed:
feat(account fragment): implemented marked for selfdestruct

lorenzogentile404 committed:
fix(self destruct time): renaming

lorenzogentile404 committed:
fix(account fragment): fix account fragment resolve post transaction

letypequividelespoubelles committed:
feat: implement the refund business

letypequividelespoubelles committed:
Merge branch 'feat/issue-746/linea-transaction-class' of github.com:Consensys/linea-arithmetization into feat/issue-746/linea-transaction-class

OlivierBBB committed:
getting rid as much as possible of one of the DomSubStampsFragment.standardDomSubStamps(...) methods

lorenzogentile404 committed:
feat(selfdestruct): added draft of contracts for testing

letypequividelespoubelles committed:
fix(OOB:RDC): typo in StackArg

letypequividelespoubelles committed:
typo

lorenzogentile404 committed:
feat(selfdestruct): fixed contracts for testing using 0.8.0+commit.c7dfd78e compiler version and VM London Environment

letypequividelespoubelles committed:
fix(hub): return data + halting instruction fixes and refactoring

letypequividelespoubelles committed:
Merge branch 'feat/issue-746/linea-transaction-class' of github.com:Consensys/linea-arithmetization into feat/issue-746/linea-transaction-class

OlivierBBB committed:
fix(defers): ChildContextEntryDefers + other

- CREATE2 writes to ROM conditionnally to not failure condition
- single "context (fragment) initialization" method
- lots of renaming

OlivierBBB committed:
fix(constraints): new hub constraints and columns + CALL scenarios

letypequividelespoubelles committed:
feat(CALL): wip

letypequividelespoubelles committed:
feat(CALL): done with exceptionnal

OlivierBBB committed:
feat(aborted calls): DONE !!!!!1!

OlivierBBB committed:
fix: added missing files ...

letypequividelespoubelles committed:
feat(EOA CALL): DONE !

letypequividelespoubelles committed:
perf(CALL): specify max nb of fragment

letypequividelespoubelles committed:
add conversions + fix Trace.java issue

lorenzogentile404 committed:
feat(call): were they both decemented?

letypequividelespoubelles committed:
feat: add ContextExitDefers registers

letypequividelespoubelles committed:
CallFrame is better

lorenzogentile404 committed:
feat(call): added ContextExitDefer interface

letypequividelespoubelles committed:
refactoring name of schedule defers

letypequividelespoubelles committed:
feat: trigger resolving ContextExitDefers

lorenzogentile404 committed:
feat(call): implement ContextExitDefer

lorenzogentile404 committed:
ras

lorenzogentile404 committed:
docs(hub): added comment for traceEndTransaction

letypequividelespoubelles committed:
add reverStamp shortcut

OlivierBBB committed:
fix(call): wip for SMC

letypequividelespoubelles committed:
refactor ras

OlivierBBB committed:
ras

OlivierBBB committed:
fix(call): various

letypequividelespoubelles committed:
clean-up

OlivierBBB committed:
feat(precompiles): first commit

letypequividelespoubelles committed:
bye bye old call ;(

letypequividelespoubelles committed:
feat(precompiles): precompile subsection work + the ADDRESS_TO_PRECOMPILE map

letypequividelespoubelles committed:
Merge branch 'arith-dev' into feat/issue-746/linea-transaction-class

letypequividelespoubelles committed:
tracing of precompile scenario fragment + cleaning scenario package

letypequividelespoubelles committed:
Merge branch 'feat/issue-746/linea-transaction-class' of github.com:Consensys/linea-arithmetization into feat/issue-746/linea-transaction-class

letypequividelespoubelles committed:
ras

letypequividelespoubelles committed:
ras

OlivierBBB committed:
feat(precompile subsection): generic methods wip

OlivierBBB committed:
fix(mmuCall): split the SHA2/RIPEMD mmu calls in 3

OlivierBBB committed:
feat(test): simple test to track the triggering of Besu's tracer interface for EOA calls

letypequividelespoubelles committed:
finish all common CALL, finish common Precompile, systematically add the messageFrame to the callFrame

letypequividelespoubelles committed:
wip change how we trigger the MMU

OlivierBBB committed:
fix(MmuCall): various

- several instructions were fixed to not defer their mmuCall's to post
transaction
- complete revision of RETURN's

OlivierBBB committed:
fix(accountSection): fixed it

AccountSection didn't care about
- exceptions
and didn't handle rollbacks the right way

OlivierBBB committed:
fix(AccountSection): missed stuff

OlivierBBB committed:
fix(ExtCodeCopySection): fixed it AFAICT

letypequividelespoubelles committed:
TRM is trigger by hand for instructions

letypequividelespoubelles committed:
ras, clean & spotless

OlivierBBB committed:
rasw

OlivierBBB committed:
ras

OlivierBBB committed:
fix(oob): OobCall slight redesign + SHA2 processing wip

OlivierBBB committed:
feat(sha2): first precompile down :)

OlivierBBB committed:
spotless

letypequividelespoubelles committed:
feat(PRC): inverse logic where the prc limits are triggered by the call to a prc data module. All done except for KECCAK

letypequividelespoubelles committed:
Merge branch 'arith-dev' into feat/issue-746/linea-transaction-class

letypequividelespoubelles committed:
implement keccak count

OlivierBBB commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/shakiradata/ShakiraData.java:
```diff
@@ -40,6 +41,7 @@ public class ShakiraData implements Module {
   private final ShakiraDataComparator comparator = new ShakiraDataComparator();
 
   private final Sha256Blocks sha256Blocks;
+  private final Keccak keccak;
```
That was where keccak was ignored previously ?

letypequividelespoubelles replied:
it was a TODO for one day, yes

OlivierBBB commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/shakiradata/ShakiraData.java:
```diff
@@ -76,8 +78,7 @@ public void call(final ShakiraDataOperation operation) {
       case SHA256 -> sha256Blocks.addPrecompileLimit(operation.inputSize());
-      case KECCAK -> { // TODO
-      }
+      case KECCAK -> keccak.addPrecompileLimit(operation.inputSize());
```
KECCAK isn't a precompile. I guess I'm confused why the KECCAK case would fall under a switch case like so
```java
  switch (operation.precompileType()) {
    case ...
}
```

OlivierBBB replied:
I guess there should be a different type of path to calling SHAKIRA or at least a renaming of the `precompileType()` method. I guess `HashType` / `HashFunction` would make more sense.

letypequividelespoubelles replied:
true, it's not a precompile at all. We should rename `precompileType` to `shakiraType` ?

OlivierBBB replied:
I would just go with `hashType`

letypequividelespoubelles committed:
feat: trigger SHAKIRA for SHA3

letypequividelespoubelles committed:
Merge branch 'arith-dev' into feat/issue-746/linea-transaction-class

letypequividelespoubelles committed:
fix(CREATE): trigger SHAKIRA and make it as the CALLs

letypequividelespoubelles committed:
fix: debug java issues : no MessageFrame given when entering Context

letypequividelespoubelles committed:
fix java issues about initiamizing frame

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix create failure case

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
feat add identity prc subsection

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
feat modexp PRC subsection

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
Merge branch 'arith-dev' into feat/issue-746/linea-transaction-class

letypequividelespoubelles committed:
fix: reject tx when arg byte size > 512

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
Merge branch 'feat/issue-746/linea-transaction-class' of github.com:Consensys/linea-tracer into feat/issue-746/linea-transaction-class

OlivierBBB committed:
feat(precompiles): wip for ECDATA precompiles

letypequividelespoubelles committed:
make it compile again + trivial bug fixes

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix modexp prc subsection (ugly), fixed OOB

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix: don't trigger resolveRollback at every context ReEntry !

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

lorenzogentile404 committed:
fix(modexpsubsection): use modexp oob calls instead of common precompile oob call

letypequividelespoubelles committed:
add check

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
minor cleaning

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

OlivierBBB committed:
ras

OlivierBBB committed:
fix(modexp): FAILURE_KNOWN_TO_RAM is the only failure mode for MODEXP

OlivierBBB committed:
feat(tests): trivial tests for for ECRECOVER and MODEXP

OlivierBBB committed:
fix(account snapthots): amended the canonical method to handle account not in the state

OlivierBBB committed:
feat(EC precompiles): done ? + various other minor changes

OlivierBBB committed:
fix(oob): partial (?) fix for callGas in OobOperation's populateColumns method

The issue was that

    ZkTracer.gasCalculator.gasAvailableForChildCall

deducts gas from the frame and shouldn't be used otherwise we find
ourselves deducting gas twice

OlivierBBB committed:
fix(oob): just some refactoring and removing of fields

OlivierBBB committed:
feat(blake): implemented BLAKE precompile

OlivierBBB committed:
fix(postRollbackDefers): fixed nullpointer exception when resolving post rollback defers

OlivierBBB committed:
TODO: failing JumpiTest because

lorenzogentile404 committed:
test(jumpi): fixed test cases

OlivierBBB committed:
fix(CallSection): fixed null pointer exception + spotless

OlivierBBB committed:
fix(call stack pattern): fixed wrong stack stamp offset

OlivierBBB committed:
fix(tracing): fixed bug with CallScenarioFragment tracing

- also added simplistic ADD test

OlivierBBB committed:
fix(call): success bit of CALL's fix ?

Not sure if complete fix, but correct success bit for CALL's to EOA's

OlivierBBB committed:
feat(test): added simple test for STP triggered by CALL instruction

OlivierBBB committed:
fix(misc/STP): sign shenanigans when setting the STP instruction

OlivierBBB committed:
fix(stpCall): STP_GAS_MXP tracing

fixed the constraint

    call-instruction---setting-STP-instruction-parameters

OlivierBBB committed:
ras

letypequividelespoubelles commented on arithmetization/src/test/java/net/consensys/linea/zktracer/instructionprocessing/BasicAdditionTest.java:
@OlivierBBB what's the purpose of this test ? It seems a duplicate of the already existing test/.../module/add/AddTest.java

letypequividelespoubelles committed:
ras: cleaning

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix unlatching stack mechanism for CALLs and CREATEs

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix casting java issues

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix java issues

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix missing account snapshot for EOA reverted successfull call

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
Merge branch 'arith-dev' into feat/issue-746/linea-transaction-class

lorenzogentile404 committed:
docs(storage): add comment

lorenzogentile404 committed:
feat: MessageFrame tests and generalised ByteCodeRunner to add additionalAccounts (#932)

letypequividelespoubelles committed:
fix(prc): missing last context rows

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix returnData for successfull ecrecover can be empty

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix: need to give ram snapshot to MMU

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
cleaning

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix: add seen storage keys in sstore and sload section + add trivial sstore sload test

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix: resolve exit & rollback defers before exitint the calldrame

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

lorenzogentile404 committed:
feat(exp): update

lorenzogentile404 committed:
Merge branch 'arith-dev' into feat/issue-746/linea-transaction-class

letypequividelespoubelles committed:
fix: missing call to ROMLEX for CREATEs

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
add TODOs for @Lorenzo

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

lorenzogentile404 committed:
feat(ecdata): update trigger using callEcData instead tracePreOpCode

lorenzogentile404 committed:
fix(ecdata): remove tracePreOpCode completely

letypequividelespoubelles committed:
fix: move the triggering of ecdata after the check that the call is successfull

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix: delete ecdata from signals

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix deployment status for CREATE's in RomLex

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix(mmu/prprc/anyToRamWithPadding): need at row 3 results of rows 5 & 6

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
Merge branch 'arith-dev' into feat/issue-746/linea-transaction-class

letypequividelespoubelles committed:
Merge branch 'arith-dev' into feat/issue-746/linea-transaction-class

letypequividelespoubelles committed:
fix(stp): trace all columns

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

FlorianHuc committed:
initialize preOpcodeCreateeSnapshot before checking aborts as it's required

letypequividelespoubelles committed:
fix: mix between CREATE and CREATE2 in stack rows, thank you Florian

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
Merge branch 'arith-dev' into feat/issue-746/linea-transaction-class

lorenzogentile404 committed:
feat(oob): add MAX_NONCE business

letypequividelespoubelles committed:
Merge branch 'arith-dev' into feat/issue-746/linea-transaction-class

FlorianHuc committed:
fix toy account storage check.

letypequividelespoubelles committed:
fix(mmu): any to ram with padding constraints (and tracing) fixes

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
Merge branch 'feat/issue-746/linea-transaction-class' of github.com:Consensys/linea-tracer into feat/issue-746/linea-transaction-class

FlorianHuc committed:
revert toy account storage check.

letypequividelespoubelles committed:
fix(modexp sub section): call OOB at preOpcode, not contextReEntry

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

FlorianHuc committed:
Merge remote-tracking branch 'origin/feat/issue-746/linea-transaction-class' into feat/issue-746/linea-transaction-class

letypequividelespoubelles committed:
Merge branch 'feat/issue-746/linea-transaction-class' of github.com:Consensys/linea-tracer into feat/issue-746/linea-transaction-class

letypequividelespoubelles committed:
fix: it's adding the oob call to the imc fragment that triggers OOB

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix keccak calling twice SHAKIRA + no need to reorder SHAKIRA op

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix(blake): trigger blakeModexp for Blake2f

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

lorenzogentile404 committed:
fix(selfdestruct): ensure contracts are compatible with London EVM

lorenzogentile404 committed:
fix(selfdestruct): update bytecode for tests

FlorianHuc committed:
use a more realistic gas price.

FlorianHuc committed:
use a more realistic gas limit.

letypequividelespoubelles committed:
fix recipient account in init phase for deployement tx

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
Merge branch 'arith-dev' into feat/issue-746/linea-transaction-class

letypequividelespoubelles committed:
test: more gas available

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix ras

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix extcodecopying empty account

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
clean: delete duplicate when creating section

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

FlorianHuc committed:
Fix NPE + class cast exception

letypequividelespoubelles committed:
ugly fix for root callDataCopy in a root context with empty payload

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix: create fictious CN even for empty call data

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix getting CFI for empty extCodeCopy

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

lorenzogentile404 committed:
fix(oob): debug basicModexpTest

lorenzogentile404 committed:
docs(oob): remove TODO

lorenzogentile404 committed:
docs(oob): modexp lead

letypequividelespoubelles committed:
fix identity prc mmuCall

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
Merge branch 'feat/issue-746/linea-transaction-class' of github.com:Consensys/linea-tracer into feat/issue-746/linea-transaction-class

OlivierBBB committed:
fix(ModexpMetadata): rawLeadingWord fix

OlivierBBB committed:
fix: fixed right zero padding + renaming to "ModexpMetaData"

OlivierBBB committed:
fix(testModexpLogSingleCase): made test pass

removed PrecompileInvocation from ModexpLogExpCall class

OlivierBBB committed:
fix: get "testMxpRandomAdvanced" working again

OlivierBBB commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/precompiles/ModexpMetadata.java:
```diff
@@ -121,11 +121,11 @@ public EWord rawLeadingWord() {
             && nExponentBytesInCallData < WORD_SIZE);
     StringBuilder callDataHexString = new StringBuilder(callData.slice(exponentOffsetInCallData, nExponentBytesInCallData).toHexString());
-    for (int i = 0; i < nExponentBytesInCallData; i++) {
+    for (int i = nExponentBytesInCallData; i < WORD_SIZE; i++) {
```
@FlorianHuc @letypequividelespoubelles @powerslider what is the canonical way of doing this right padding ?

OlivierBBB replied:
Oh I may be thinking of

```java
 static Bytes32 rightPad(Bytes value)
```

OlivierBBB replied:
Should be fixed with https://github.com/Consensys/linea-tracer/pull/748/commits/1b5489df7b353775541c90ca2faeb2ec6df2bc9c

OlivierBBB committed:
fix(testSeveralKeccaks): test passes now

OlivierBBB committed:
feat: new "is the beef dead yet ?" tests for KECCAK

OlivierBBB committed:
fix(rawLeadingWord): simplified the rawLeadingWord() method

letypequividelespoubelles committed:
spotless + manually update shakira Trace.java + constraint update

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
constraint update

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/KeccakSection.java:
```diff
@@ -50,10 +50,12 @@ public KeccakSection(Hub hub) {
     if (triggerMmu) {
       final MmuCall mmuCall = MmuCall.sha3(hub);
       imcFragment.callMmu(mmuCall);
+      long offset = hub.messageFrame().getStackItem(0).toLong();
```
don't use .toLong, but Words.clampedToLong

letypequividelespoubelles replied:
fixed https://github.com/Consensys/linea-tracer/pull/748/commits/c6d8ef46ddbfabe880f8ff94993dd0543e9dbcd9

letypequividelespoubelles committed:
fix converting stack item to long

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix(hubStamp): update stamp when creating a section

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix set returner info while returning from a precompile call

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

OlivierBBB committed:
feat(trm): a new trimming test for the HUB

powerslider committed:
fix: invalid code prefix MMU instruction patch

lorenzogentile404 committed:
fix(constraints): update to fix failing oob tests--no-verify

lorenzogentile404 committed:
feat(gas): update ACDC

OlivierBBB committed:
feat(modexp): new tests for MODEXP

letypequividelespoubelles committed:
style: spotless

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

lorenzogentile404 committed:
feat(constraints): update

lorenzogentile404 committed:
feat(constraints): update

letypequividelespoubelles committed:
fix: inverted logic of triggering of mmu in callDataLoad

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

lorenzogentile404 committed:
fix(gas): move gas in trace-files.gradle

lorenzogentile404 committed:
fix(gas): update implementation after Trace.java update

lorenzogentile404 committed:
docs(gas): add comment

lorenzogentile404 committed:
fix(ecdata): fill success bit correctly

FlorianHuc committed:
change runner name

FlorianHuc committed:
Merge remote-tracking branch 'origin/feat/issue-746/linea-transaction-class' into feat/issue-746/linea-transaction-class

OlivierBBB committed:
fix(MARKED_FOR_SELFDESTRUCT): fixed typo

OlivierBBB committed:
fix(modexp): use safe slice method to extract bbs, ebs and mbs

OlivierBBB committed:
fix(modexp): extract the correct "rawLeadingWord"

OlivierBBB committed:
ras

OlivierBBB committed:
spotless

OlivierBBB committed:
fix(oob): typo

OlivierBBB committed:
fix(rollback): REVERT now also triggers the rollback mechanism

OlivierBBB committed:
fix(mmu): we only commit those MmuOperations that say "traceMe()"

OlivierBBB committed:
test(mmu): added tests for MMU

Note. There is a bug somewhere, REVERT opcodes don't trigger the MMU ...

OlivierBBB committed:
fix(mmu): line count now discards operations that don't get traced

OlivierBBB committed:
feat(mxp): some extra tests

they don't shed light on the issue ...

OlivierBBB committed:
feat: traceMe() method for MmuOperation

OlivierBBB commented:
@letypequividelespoubelles could you please review
- https://github.com/Consensys/linea-tracer/pull/748/commits/f249e99f22c6b9b2e747b42393ce8fdb477d359c
- https://github.com/Consensys/linea-tracer/pull/748/commits/7cdec09e7de89113cebe7a3c358c5e27f03b9447

?

letypequividelespoubelles committed:
spotless + fix untraced MMU op in mmio

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles commented:
> @letypequividelespoubelles could you please review
> 
>     * [f249e99](https://github.com/Consensys/linea-tracer/commit/f249e99f22c6b9b2e747b42393ce8fdb477d359c)
> 
>     * [7cdec09](https://github.com/Consensys/linea-tracer/commit/7cdec09e7de89113cebe7a3c358c5e27f03b9447)
> 
> 
> ? This doesn't solve `testRandomMxpInstructionsFollowedByTriggeringRoob` but it seems like it was a bug (AFAICT.)

fixed & review https://github.com/Consensys/linea-tracer/pull/748/commits/9658d9318d7d900802c66d3b824ef87191d7eed7
 
I added skipping it for MMIO too (both tracing & line counting), and I prefer to modify the `lineCount` of the `mmuOperation` instead of what you've done in https://github.com/Consensys/linea-tracer/commit/7cdec09e7de89113cebe7a3c358c5e27f03b9447

letypequividelespoubelles committed:
Merge branch 'arith-dev' into feat/issue-746/linea-transaction-class

letypequividelespoubelles committed:
fix make it build

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix successBit in invalidCodePrefix MmuCall

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix: aborting calls defers at childContextEntry

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles committed:
fix postOpcode acc snapshot for aborting calls

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/ecdata/EcDataOperation.java:
```diff
@@ -122,34 +120,39 @@ public class EcDataOperation extends ModuleOperation {
+    Preconditions.checkArgument(precompileFlag.isEcdataPrecompile(), "invalid EC type");
+
+    this.precompileFlag = precompileFlag;
+    final int paddedCallDataLength = this.precompileFlag == PRC_ECMUL ? 96 : 128;
```
should be constant of the class

letypequividelespoubelles commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/ecdata/EcData.java:
```diff
@@ -97,17 +87,38 @@ public void commit(List<MappedByteBuffer> buffers) {
       // TODO: temporary hack, we should have only successful EcData Operations, will be fixed in PR
-      // #748
+      //  #748
       if (op.successBit()) {
```
@lorenzogentile404 @OlivierBBB  could you check this plz ? 
I think that now, when we call EcData, we trace it. 
Could be problematic to keep it like this, as for ecRecover, we have `successBit = !recoveredAddress.isZero();`

letypequividelespoubelles commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/ecdata/EcData.java:
```diff
@@ -97,17 +87,38 @@ public void commit(List<MappedByteBuffer> buffers) {
+      case PRC_ECMUL -> ecMulEffectiveCall.addPrecompileLimit(1);
+      case PRC_ECRECOVER -> ecRecoverEffectiveCall.addPrecompileLimit(1);
+      case PRC_ECPAIRING -> {
+        // TODO @Lorenzo
```
plz @lorenzogentile404 ;)

letypequividelespoubelles commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/Hub.java:
```diff
@@ -1203,403 +809,284 @@ private void handleCreate(Address target) {
-              this, this.currentFrame(), ContextFragment.executionEmptyReturnData(callStack));
-    }
+  /**
+   * 0x 0000000000000000000000000000000000000000000000000000000000000020
```
@OlivierBBB What is it ? Was it for debuging purpose and can now be deleted ?

OlivierBBB replied:
It seems like very old stuff (debugging return data, so maybe as old as Tbilissi ?)

letypequividelespoubelles committed:
cleanup

Signed-off-by: Francois Bojarski <francois.bojarski@consensys.net>

letypequividelespoubelles reviewed (commented):
Finally ;)

lorenzogentile404 committed:
docs(hub): add comment related to creator nonce

OlivierBBB commented on arithmetization/src/main/java/net/consensys/linea/zktracer/module/ecdata/EcDataOperation.java:
```diff
@@ -122,34 +120,39 @@ public class EcDataOperation extends ModuleOperation {
+    this.precompileFlag = precompileFlag;
+    final int paddedCallDataLength = this.precompileFlag == PRC_ECMUL ? 96 : 128;
+    if (callData.size() < paddedCallDataLength) {
+      this.data = leftPadTo(callData, paddedCallDataLength);
```
This ought to be a `rightPadTo`, no ?

lorenzogentile404 committed:
fix(ecdata): partial fix

OlivierBBB committed:
fix(ecdata): call data has to be right padded + 100% valid padded call data length

# Conflicts:
#	arithmetization/src/main/java/net/consensys/linea/zktracer/module/ecdata/EcDataOperation.java

OlivierBBB committed:
fix(ecdata): ensure that EcData has nonempty call data + renaming

lorenzogentile404 committed:
spotless

OlivierBBB committed:
feat(ecrecover): ultra simple failing test

lorenzogentile404 committed:
fix(ecdata): re-enable tracing and line counting disregarding success bit value

lorenzogentile404 committed:
spotless

OlivierBBB reviewed (approved):
LGTM
